contracts/src/spaces/facets/tipping/TippingBase.sol

```diff
@@ -18,6 +18,12 @@ library TippingBase {
   struct Layout {
     EnumerableSet.AddressSet currencies;
     mapping(uint256 tokenId => mapping(address currency => uint256 amount)) tipsByCurrencyByTokenId;
+    mapping(address currency => uint256 amount) totalTipAmountByCurrency;
+    mapping(address currency => uint256 count) totalTipCountByCurrency;
+    mapping(address user => mapping(address currency => uint256 amount)) tipsReceivedByCurrency;
+    mapping(address user => mapping(address currency => uint256 amount)) tipsSentByCurrency;
+    mapping(address user => mapping(address currency => uint256 count)) tipsReceivedCountByCurrency;
+    mapping(address user => mapping(address currency => uint256 count)) tipsSentCountByCurrency;
```

Comment:
I'd suggest combining the mappings with
```solidity
  struct CurrencyStat {
    uint128 amount;
    uint128 count;
  }

  struct UserStat {
    uint128 receivedAmount;
    uint128 receivedCount;
    uint128 sentAmount;
    uint128 sentCount;
  }

  struct Layout {
    mapping(address currency => CurrencyStat) statByCurrency;
    mapping(address user => mapping(address currency => UserStat)) statByUserAndCurrency;
  }
```
For less verbosity and more efficiency.